### DISCUSSION TITLE
PDE with small coefficient

### DISCUSSION THREAD

**Discussion by bahrami4444** [lululxvi/deepxde#1921] [General]
Thanks to Dr. Lu @lululxvi  for providing this simple and powerful software.

I want to model the following equations in DeepXDE.
![111](https://github.com/user-attachments/assets/189fe23f-7323-4f15-b54d-98ab66080358)

These equations are coupled and are coded as follows:

```
def PDE(Input, Output):
    
    u_yy = dde.grad.hessian(Output, Input, i=0, j=0, component=0)
    u_ty = dde.grad.hessian(Output, Input, i=0, j=1, component=0)
    p_yy = dde.grad.hessian(Output, Input, i=0, j=0, component=1)
    p_y = dde.grad.jacobian(Output, Input, i=1, j=0)
    p_t = dde.grad.jacobian(Output, Input, i=1, j=1)
    
    return [(λ + 2 * μ) * u_yy - α * p_y, α * u_ty + φ * β * p_t - κ * p_yy]
```

My problem is that the beta coefficient is too small (4.4e-4), and therefore the neural network cannot train properly. When I set beta to 1, the output of the neural network becomes logical (but incorrect). The figure below shows the logical answer:

![1](https://github.com/user-attachments/assets/47482cad-00a2-4dc9-ade6-ed6d12ea7219)

And this figure represents the case when the real value of beta is applied:

![20200](https://github.com/user-attachments/assets/991d0860-9fc2-44d9-9a65-52629b30f89b)

**Does anyone know how to fix this problem?**

Using a dimensionless governing equation is not a good solution because, in such equations, t and z (space and time) become dimensionless, but the value of the beta coefficient remains small.
Scaling all coefficients also makes the other coefficients either too small or too large.
I also significantly reduced the learning rate and increased the number of layers, but the problem was not solved.

Please guide me.
Thanks.
